### PR TITLE
FRT-4-5 created page for adding and saving notes

### PR DESCRIPTION
**Pull request by AnastiiaP**

### CODE REVIEW

File: Front/addsavenote.cpp [hunk 1]
```diff
@@ -0,0 +1,146 @@
+#include "addsavenote.h"
+#include "ui_addsavenote.h"
+
+
+using namespace std;
```
Comment (VictorApopiy) [inline Front/addsavenote.cpp]: Please use "std::" instead of opening all namespace

File: Front/addsavenote.cpp [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+    delete ui;
+}
+
+struct Note
+{
+    QString sTitle;
+    QString sDescription;
+    vector <QString> vsKeywords;
+    QString sCategory;
+    QString sAccessUser;
```
Comment (VictorApopiy) [inline Front/addsavenote.cpp]: Add m_ prefix to struct/class members

File: Front/addsavenote.cpp [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+};
+
+void MainWindow::CreateJson(const QString &path)
+{
+    QRegularExpression rx("(\\\n|\\:|\\;)");
+    QString sKeywordsText = ui->KeywordsTEdit->toPlainText();
+    QStringList myStringList = sKeywordsText.split(rx);
+    vector <QString> vsKeywords;
+    QJsonObject keywords;
+    for(int i=0;i<myStringList.size();i++)
```
Comment (VictorApopiy) [inline Front/addsavenote.cpp]: is is better to use pre increment ++i

File: Front/addsavenote.cpp [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+        file.close();
+    }
+    else
+    {
+        cout << "file open failed: " << path.toStdString() << endl;
+    }
+}
+
+
+void ReadJson(const QString &path)
```
Comment (VictorApopiy) [inline Front/addsavenote.cpp]: it is better to make this func bool

File: Front/addsavenote.cpp [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+    {
+        QByteArray bytes = file.readAll();
+        file.close();
+
+        QJsonParseError jsonError;
+        QJsonDocument document = QJsonDocument::fromJson( bytes, &jsonError );
+        if( jsonError.error != QJsonParseError::NoError )
+        {
+            cout << "fromJson failed: " << jsonError.errorString().toStdString() << endl;
+            return ;
```
Comment (VictorApopiy) [inline Front/addsavenote.cpp]: and return false

File: Front/addsavenote.cpp (hunk 1 above)
Comment (AnastiiaP) [reply Front/addsavenote.cpp]: Fixed

File: Front/addsavenote.cpp (hunk 2 above)
Comment (AnastiiaP) [reply Front/addsavenote.cpp]: Fixed

File: Front/addsavenote.cpp (hunk 4 above)
Comment (AnastiiaP) [reply Front/addsavenote.cpp]: Fixed

File: Front/addsavenote.cpp (hunk 3 above)
Comment (AnastiiaP) [reply Front/addsavenote.cpp]: Fixed

File: Front/addsavenote.cpp (hunk 5 above)
Comment (AnastiiaP) [reply Front/addsavenote.cpp]: Fixed

**Review by VictorApopiy — APPROVED** [review]

**Pull request merged**